presto-main/src/main/java/com/facebook/presto/operator/scalar/annotations/ParametricScalarImplementation.java

```diff
@@ -325,6 +385,9 @@ private void parseArguments(Method method)
                     boolean hasNullFlag = false;
                     if (method.getParameterCount() > (i + 1)) {
                         Annotation[] parameterAnnotations = method.getParameterAnnotations()[i + 1];
+                        if (Stream.of(annotations).anyMatch(BlockPosition.class::isInstance)) {
```

Comment:
This `if` should not be in `if (method.getParameterCount() > (i + 1))`. Instead, you would want to move it out so that you check for this annotation regardless of whether a next argument exists. In addition, you would want to assert that 1) the next argument exists, 2) the next argument has `@BlockIndex` annotation.